com.eclipsesource.modelserver.emf/src/main/java/com/eclipsesource/modelserver/emf/launch/ModelServerLauncher.java

```diff
@@ -57,7 +59,9 @@ public void start() {
 	}
 
 	protected void run() {
-		injector.getInstance(ModelServerStartup.class).boot(EntryPointType.REST, this.args);
+		int port = args.length > 0 ? Integer.parseInt(args[0].split("=")[1]) : DEFAULT_JAVALIN_PORT;
```

Comment:
This causes an uncatched exceptions (`ArrayIndexOutOfBoundsException`, `NumberFormatException`) if the argument doesn't have the right format. (no `=` or the right-hand side of `=` cannot be cast to an integer)

We should catch this and print some information about invalid arguments and the correct usage.

In addition, the --port prefix is currently not checked. IMHO it should either be checked if the argument starts with "--port" or the prefix should be omitted.

Reply:
Thanks for testing, that's a good point!